docs/kind-example-config.yaml

```diff
@@ -3,8 +3,29 @@ apiVersion: kind.sigs.k8s.io/v1alpha1
 kind: Config
 # number of nodes in the cluster (currently only 1 is supported)
 numNodes: 1
-# template for kubeadm config, "" -> the default template
-kubeadmConfigTemplate: ""
+# these patches are applied to the generated kubeadm config
+# using `kustomize build` where the kustomization.yaml looks something like:
+# resources:
+# - generated-config.yaml
```

Comment:
From a completely new users perspective, I think this additional boilerplate inside of our own examples might get a bit confusing. Could we either double # it (in an attempt to make it clearer that these are *not* valid `kind` directives), or otherwise explain what `generated-config.yaml` is (assuming we think it's important for the user to know).

To be honest, I would rather that kustomize is treated as an implementation detail. A sentence like "`kubeadmConfigPatches` allows you to specify custom configuration to kubeadm that is not otherwise exposed by kind. Kind will layer these patches on top of its own, generated, based configuration".

Just thinking here, that a lot of users will not have touched kustomize at all, and so could easily not recognise what's going on here.

Maybe we could then move some of the more detailed information into `custom-config.md` doc 😄

Reply:
removed this, even if they've touched kustomize we're wrapping things a bit to make all the config inline